File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.ts
function longestSubarray(nums: number[]): number {
    let highest = 0;
    let count = 0, highestCount = 0;

    for(let i = 0; i < nums.length; i++) {
        if(highest === nums[i]) {
            count++
            highestCount = Math.max(highestCount, count)
        } else {
            if(nums[i] > highest) {
                highest = nums[i]
                count = 1;
                highestCount = count;
            } else {
                count = 0
            }
        }
    }
    return highestCount;
};

File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let mut highest = nums[0];
        let mut max = 1;
        let mut count = 1;
        for i in 1..nums.len() {
            if highest == nums[i] {
                count += 1;
                max = max.max(count);
            } else {
                if nums[i] > highest {
                    highest = nums[i];
                    count = 1;
                    max = 1;
                } else {
                    count = 0;
                }
            }
        }
        max
    }
}